addons/website_slides/static/src/js/activity.js

```diff
@@ -42,4 +42,178 @@ function applyInclude(Activity) {
+
+    registerClassPatchModel('mail.activity', 'website_slides/static/src/js/slide_models.js', {
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        /**
+         * @override
+         */
+        convertData(data) {
+            const data2 = this._super(data);
+            if ('request_partner_id' in data && 'res_id' in data) {
+                if (!data.request_partner_id) {
+                    data2.slide = [['unlink-all']];
+                } else {
+                    data2.slide = [
+                        ['insert', {
+                            id: data.res_id,
+                            partnerId: data.request_partner_id[0],
+                        }],
+                    ];
+                }
+            }
+            return data2;
+        },
+    });
+
+    registerFieldPatchModel('mail.activity', 'website_slides/static/src/js/slide_models.js', {
+        slide: one2one('slides.slide', {
+            inverse: 'activity',
+        }),
+    });
+
+    function factory(dependencies) {
+
+        class Slide extends dependencies['mail.model'] {
+            //----------------------------------------------------------------------
+            // Public
+            //----------------------------------------------------------------------
+
+            /**
+             * Grant the access for the current `Course`.
+             */
+            async grantAccess() {
+                await this.async(() => this.env.services.rpc({
+                    model: 'slide.channel',
+                    method: 'action_grant_access',
+                    args: [this.id, this.partnerId],
+                }));
+                if (this.activity) {
+                    this.activity.delete();
+                }
+                this.delete();
+            }
+            /**
+             * Refuse the access for the current `Course`.
+             */
+            async refuseAccess() {
+                await this.async(() => this.env.services.rpc({
+                    model: 'slide.channel',
+                    method: 'action_refuse_access',
+                    args: [this.id, this.partnerId],
+                }));
+                if (this.activity) {
+                    this.activity.delete();
+                }
+                this.delete();
+            }
+
+            //----------------------------------------------------------------------
+            // Private
+            //----------------------------------------------------------------------
+
+            /**
+             * @override
+             */
+            static _createRecordLocalId(data) {
+                return `${this.modelName}_${data.id}`;
+            }
+
+        }
+
+        Slide.fields = {
+            activity: one2one('mail.activity', {
```

Comment:
`one2one` looks like a mistake here. There could be multiple request/activities for a single slide I believe.